### ISSUE TITLE
Migrating from Localhost to Live MySQL 5.5 causes error

### ISSUE THREAD

**Issue by hrsetyono** [wp-sync-db/wp-sync-db#116]
This is not an issue with this plugin but with WP v4.7 that uses `utf8mb4_unicode_520_ci` collation that's not supported in MySQL 5.5.

So if your localhost is 5.6+ (which should be default installation) but your server is 5.5, you will get this error:

```
WordPress database error: [Can't find file: './your_site/_mig_wp_commentmeta.frm' (errno: 2)]
RENAME TABLE `_mig_wp_commentmeta` TO `wp_commentmeta`
```

Manual export-import using PHPMyAdmin also cause error, unless you choose "MySQL40" in "Format-specific-options > Maximize output compatibility with:"

If the plugin can has the setting for **compatibility output**, that would be awesome.

**Comment by nicooprat** [comment]
Same here. Do you have any workaround for now? Like a MySQL parameter to config that locally?

**Comment by hrsetyono** [comment]
HI @nicooprat, I solved this by creating a new MySQL database with `utf8mb4_unicode` encoding (usually the default is utf8). You can't change existing one, so must create from scratch.

I'm not sure whether all hosting support this, but it does on Webfaction.

**Comment by max-sixblade** [comment]
I believe this is not mysql 5.6 to 5.5 problem but wordpress version. I believe wordpress uses utf8mb4 by default since 4.2. As for mysql... Recently I've updated my localhost mysql to 5.6 and now I'm a bit worried about transferring websites to live servers because mysql v 5.6 is rarely used rarely by hosting companies.

**Comment by hrsetyono** [comment]
@max-sixblade The issue is indeed on WP version since I never get that error when using 4.6. But recently my host upgraded MySQL to 5.6 and everything works great without special config.

Which mean that error only come out when you're using WP 4.7+ and your server MySQL is 5.5 or lower

**Comment by max-sixblade** [comment]
@hrsetyono This might be another instance of the same problem. Here is an interesting topic about it https://make.wordpress.org/core/2015/04/02/the-utf8mb4-upgrade/
The problem is since 4.2 WP is using utf8mb4 by default. So if your local dev is supporting utf8mb4 but production server does not.. here is where problems appear. Here is a link to the script that converts utf8mb4-to-utf8 (I think data loss still might appear but this never happened to me ).. http://ben.lobaugh.net/blog/201740/script-to-convert-mysql-collation-from-utf8mb4-to-utf8 ..it helped me several times where at production server there where no utf8mb4 available.

**Comment by warudin** [comment]
I'm having this problem as well when migrating from localhost to a server. In this case there are some tables with this collation: 'utf8mb4_unicode_520_ci'.

If I change this collation to something like 'utf8_unicode_ci' I am able to migrate the database to a server with WP Sync DB.

Why the tables are using this weird format by default, I'm not sure. I'd love to have a more simple solution but this is the best so far.

**Comment by hrsetyono** [comment]
@warudin Yes, seeing as the project is no longer active, I believe that's the simplest solution we got.

**Comment by warudin** [comment]
Right, I wasn't sure if the project wasn't in development anymore. That's a pity, because I was pretty happy with the prospect of being able to synchronise databases back- and forward from development, staging and live environments!

**Comment by petrchutny** [comment]
Similar issue here, changing collations everywhere to utf8mb4_unicode_ci didn't help.

**Comment by numediaweb** [comment]
Your databases are using different versions of MySQL, one version supports 'utf8mb4_unicode_ci', while the other doesn't: you need to upgrade your mysql client:
```
sudo apt-get update
sudo apt-get upgrade
sudo apt-get install mysql-client-5.6 mysql-client-core-5.6 mysql-server-5.6
```

**Comment by petrchutny** [comment]
Thanks a lot for this reply @numediaweb , next time it happens, I know where to look :)
